scio-managed/src/main/scala/com/spotify/scio/managed/ManagedIO.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spotify.scio.managed
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.io.{EmptyTap, EmptyTapOf, ScioIO, Tap, TapT}
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.managed.Managed
+import org.apache.beam.sdk.schemas.Schema
+import org.apache.beam.sdk.values.{PCollectionRowTuple, Row}
+import scala.jdk.CollectionConverters._
+
+final case class ManagedIO(ioName: String, config: Map[String, Object]) extends ScioIO[Row] {
+  override type ReadP = ManagedIO.ReadParam
+  override type WriteP = ManagedIO.WriteParam
+  override val tapT: TapT.Aux[Row, Nothing] = EmptyTapOf[Row]
+
+  private lazy val _config: java.util.Map[String, Object] = {
+    // recursively convert this yaml-compatible nested scala map to java map
+    // we either do this or the user has to create nested java maps in scala code
+    // both are bad
+    def _convert(a: Object): Object = {
+      a match {
+        case m: Map[_, _] =>
+          m.asInstanceOf[Map[_, Object]].map { case (k, v) => k -> _convert(v) }.asJava
+        case i: Iterable[_] => i.map(o => _convert(o.asInstanceOf[Object])).asJava
+        case _              => a
+      }
+    }
+    config.map { case (k, v) => k -> _convert(v) }.asJava
+  }
```

Comment:
I'm wondering if we should introduce either a config AST to ensure what is passed is YAML compatible, or maybe use lightbend [config](https://github.com/lightbend/config/blob/d279841c425e70645feacbdc03e41b3b022ad1f9/config/src/main/java/com/typesafe/config/ConfigObject.java#L81)

Reply:
The API can aslo take a yaml file location, e.g. `classpath://foo.yaml` if we wanted to support that.